src/Npgsql/TypeHandlers/CompositeHandlers/CompositeHandler.cs

```diff
@@ -90,25 +97,89 @@ public override int ValidateAndGetLength(T value, ref NpgsqlLengthCache? lengthC
             lengthCache.Set(0);
 
             // number of fields + (type oid + field length) * member count
-            var length = sizeof(int) + sizeof(int) * 2 * _memberHandlers.Length;
-            foreach (var member in _memberHandlers)
+            var memberHandlers = _memberHandlers!;
```

Comment:
Any reason for this local variable?

Reply:
Yep, bangs. Don't like them, but when I did the feature there were no attributes to mark fields.